examples/zephyr/common/Kconfig

```diff
@@ -4,6 +4,110 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+menuconfig GOLIOTH_SAMPLE_COMMON
+	bool "Common helper code shared across multiple samples"
+	help
+	  Build and link common code that is shared across Golioth samples.
+
+if GOLIOTH_SAMPLE_COMMON
+
+rsource "Kconfig.defconfig"
+
+choice GOLIOTH_AUTH_METHOD
```

Comment:
I think this should be `GOLIOTH_SAMPLE_HARDCODED_CRED_TYPE` or something along those lines. The SDK supports both PSK and Cert auth at runtime, so these symbols only apply to the hardcoded credentials module.

Reply:
> The SDK supports both PSK and Cert auth at runtime, so these symbols only apply to the hardcoded credentials module.

It shouldn't support both at runtime, at least with "production-like" configuration. Having support for all ciphersuites, even if not used, doesn't make sense. mbedTLS (or TLS in general) is taking a lot of resources and I've reduced its configuration a lot of times in the past with downstream projects, which grew up over time in terms of functionality, but there was just a limited resource pool (both Flash and RAM).

Reply:
I agree, we'll want to make it a compile time choice later. But that's not happening in this PR, and if it was, it wouldn't make sense to have these symbols defined in the examples Kconfig file. We can move the symbol and update the naming when we switch to selecting the auth type and cipher suites at build time.

Reply:
I discussed this with @sam-golioth and have moved the `GOLIOTH_AUTH_METHOD` related symbols to `ports/zephyr/Kconfig`.